src/app/manifest_manager.rs

```diff
@@ -0,0 +1,68 @@
+use failure::Error;
+use diesel;
+use diesel::prelude::*;
+use chrono::Local;
+
+use crate::{
+    db::{
+        DbConnection,
+        manifest::{
+            schema::manifest,
+            models::{Manifest, NewManifest}
+        },
+        manifest_content::{
+            schema::manifest_content,
+            models::{ManifestContent, NewManifestContent}
+        }
+    }
+};
+
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug, Default)]
+pub struct ManifestManager {}
+
+impl ManifestManager {
+    #[inline]
+    pub fn new() -> ManifestManager {
+        ManifestManager {}
+    }
+
+    pub fn create(
+        db_con: &DbConnection,
+        name: &str,
+        content: &str
+    ) -> Result<ManifestContent, Error> {
+
+        // Create new Manifest object
+        let new_manifest = NewManifest {
+            name
+        };
+
+        // Insert a new manifest in database
+        diesel::insert_into(manifest::table)
+            .values(&new_manifest)
+            .execute(db_con.as_ref())?;
+
+        // Retrieve the inserted row
+        let res : Manifest = manifest::table
```

Comment:
I don't know if diesel provide any kind of protections against this, but this looks like a data race.

If two threads run at the same time, both inserting into the db at the same time (line 41-43), they will both retrieve the same 'last id' (line 46-48), making this method weak in a multi-threading context.

I think you can use `returning()` instead of `values()` in the previous expression, returning the freshly added value.

( See [here](https://docs.diesel.rs/diesel/query_builder/struct.InsertStatement.html#method.returning) )

Reply:
Fixed.
Database migrated to postgres so `RETURNING` statement is available.